pkg/cloud/azure/services/compute/virtualmachines.go

```diff
@@ -143,83 +203,28 @@ func (s *Service) createVM(machine *actuators.MachineScope, bootstrapToken, kube
 			})
 
 			if err != nil {
-				return input, err
+				return "", err
 			}
 		}
 
-		input.StartupScript = cfg
-
 	case "node":
 		// TODO: Check for existence of node subnet & ensure NSG is attached to subnet
-
-		cfg, err := config.NewNode(&config.NodeInput{
+		var err error
+		startupScript, err = config.NewNode(&config.NodeInput{
 			CACertHash:        caCertHash,
 			BootstrapToken:    bootstrapToken,
 			LBAddress:         s.scope.Network().APIServerIP.DNSName,
 			KubernetesVersion: machine.Machine.Spec.Versions.ControlPlane,
 		})
 
 		if err != nil {
-			return input, err
+			return "", err
 		}
 
-		input.StartupScript = cfg
-
 	default:
-		return nil, errors.Errorf("Unknown node role %q", machine.Role())
-	}
-
-	// TODO: Set ssh key
-
-	resourcesSvc := resources.NewService(s.scope)
-	err = resourcesSvc.ValidateDeployment(machine.Machine, s.scope.ClusterConfig, machine.MachineConfig, input.StartupScript)
-	if err != nil {
-		return nil, fmt.Errorf("error validating deployment: %v", err)
-	}
-
-	deploymentsFuture, err := resourcesSvc.CreateOrUpdateDeployment(machine.Machine, s.scope.ClusterConfig, machine.MachineConfig, input.StartupScript)
-	if err != nil {
-		return nil, fmt.Errorf("error creating or updating deployment: %v", err)
-	}
-
-	err = resourcesSvc.WaitForDeploymentsCreateOrUpdateFuture(*deploymentsFuture)
-	if err != nil {
-		return nil, fmt.Errorf("error waiting for deployment creation or update: %v", err)
-	}
-
-	deployment, err := resourcesSvc.GetDeploymentResult(*deploymentsFuture)
-	// Work around possible bugs or late-stage failures
-	if deployment.Name == nil || err != nil {
-		return nil, fmt.Errorf("error getting deployment result: %v", err)
+		return "", errors.Errorf("Unknown node role %s", machine.Role())
 	}
-
-	vm, err := s.scope.VM.Get(s.scope.Context, machine.ClusterConfig.ResourceGroup, machine.Name(), "")
-	if err != nil {
-		return nil, err
-	}
-
-	record.Eventf(machine.Machine, "CreatedVM", "Created new %s vm with id %q", machine.Role(), vm.ID)
-	return converters.SDKToVM(vm), nil
-}
-
-// DeleteVM deletes the virtual machine.
-func (s *Service) DeleteVM(resourceGroup string, name string) (compute.VirtualMachinesDeleteFuture, error) {
-	return s.scope.VM.Delete(s.scope.Context, resourceGroup, name)
-}
-
-// MachineExists will return whether or not a machine exists.
-func (s *Service) MachineExists(machine *actuators.MachineScope) (bool, error) {
-	var err error
-	var instance *v1alpha1.VM
-
-	if machine.MachineStatus.VMID != nil {
-		instance, err = s.VMIfExists(machine.Name())
-	}
-
-	if err != nil {
-		return false, errors.Wrapf(err, "failed to lookup machine %q", machine.Name())
-	}
-	return instance != nil, nil
+	return startupScript, nil
```

Comment:
This needs to be moved into its own PR, this doesnt look like it has to do with golangci-lint

Reply:
This is needed because this is a piece of code the violates the `gocyclo` linter. I am using this PR so that the project becomes ready for a CI lint job. So I'd argue that this change is still in scope of what this PR intends to do.

Reply:
I see, that wasn't obvious to me. Thank you for the clarification